### ISSUE TITLE
Allow string values in JOSE properties in base JSON-LD context

### ISSUE THREAD

**Issue by msporny** [w3c/vc-data-model#1275]
The current base JSON-LD Context requires certain JOSE/JWK properties to be URLs by setting the `type` property to the value of `@id` (which, in JSON-LD, effectively means: "This value MUST be a URL").

For example, this exists in the current base JSON-LD Context:

```
"kid": {
    "@id": "https://www.iana.org/assignments/jwt#kid",
    "@type": "@id"
}
```

@selfissued confirmed in https://github.com/w3c/vc-data-model/pull/1259/files/2414ad34c95734ca0e8c65202a8294bf4639d189#r1309411981 that "kid" does not necessarily have to be a URI. 

There are multiple JOSE/JWK properties that follow this pattern and this issue is to track the concern that this seems to not be aligned with what the JOSE/JWK specifications say. If we are going to define JOSE/JWK properties in the base context, we should make sure that they're completely aligned w/ the JOSE/JWK specifications to increase interoperability.

**Comment by selfissued** [comment]
@msporny is correct that JOSE `kid` values can be any string.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-09-14-vcwg#section2-1) on 2023-09-14

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.1. Allow string values in JOSE properties in base JSON-LD context (issue vc-data-model#1275)

_See github issue [vc-data-model#1275](https://github.com/w3c/vc-data-model/issues/1275)._




**Brent Zundel:** manu can you comment on this issue?  

**Manu Sporny:** This came up when we were processing fix ups to the base context. We were getting it in the proper shape to go into CR. We noticed that Orie had set some of the jwk properties to let them be URL, where the JWK doesn't allow that.  
� An example is "kid". It's currently defined as "it must be a URL" in the base context.  
� JWK says that it's a string.  
� We shouldn't deviate, we should align.  

**Michael Jones:** I agree with manu that "kid" needs to be able to be any string.  

**Manu Sporny:** I can take it. It will likely be objected, though. Can someone from the vc-jose-cose world do it?  
� We should align. That's what I'd like to see.  

**Michael Jones:** I can review the PR.  

**Brent Zundel:** Assigning to both manu and selfissued.  

**Michael Jones:** manu, please ping me out of band.  

**Brent Zundel:** Any more comments? (silence).  

</details>

**Comment by msporny** [comment]
PR #1325 has been raised to address this issue. This issue will be closed once PR #1325 has been merged.

**Comment by msporny** [comment]
PR #1325 failed to achieve consensus, and it's unlikely the the current group would ever achieve consensus on such a PR, closing.
